src/payment/strategies/afterpay/afterpay-payment-strategy.ts

```diff
@@ -101,7 +104,11 @@ export default class AfterpayPaymentStrategy implements PaymentStrategy {
         return this._store.dispatch(this._paymentActionCreator.submitPayment(paymentPayload));
     }
 
-    private _redirectToAfterpay(countryName: string, paymentMethod?: PaymentMethod): void {
+    private _redirectToAfterpay(countryName: string, amount: number, paymentMethod?: PaymentMethod): void {
+        if (this._outsideLimits(amount, paymentMethod)) {
```

Comment:
please move this validation out of this method and call it first (before calling this one)